### PR TITLE
Increase station length limit to 32

### PR DESCRIPTION
**Pull request by Gymnasiast**
The station length limits seems to be completely arbitrary. I made several long stations without ill effects, making the longer trains no longer look out of place.

scr198

### CODE REVIEW

**Comment by janisozaur** [comment]
Wouldn't it rather make sense to only get enabled when the long trains cheat is in effect?

**Comment by Gymnasiast** [comment]
I wouldn't say so. The limit is completely arbitrary and does not make the game more challenging.

**Comment by Nubbie** [comment]
The limit is still a helper for new players to not build too long stations

**Comment by janisozaur** [comment]
Wasn't the limit calibrated to allowed vehicles' length?

**Comment by Gymnasiast** [comment]
No, it was always 12. (You can see that in the diff quite easily.)

**Comment by janisozaur** [comment]
Yes, I saw, but the question was more intended to sound like: Is the length of longest train allowed 12 and so is the station limit?

**Comment by IntelOrca** [comment]
I am not sure having no limit at all is a good idea. Why not just extend it to 32 or something?

**Comment by pizza2004** [comment]
I also think this should only be in effect when you disable various limits using cheats.

**Comment by Gymnasiast** [comment]
@IntelOrca But why?! It's an arbitrary limit and nothing more.

**Comment by Manuel-K** [comment]
> It adds nothing to the game.
Doesn't it? It prevents building really long stations that are only useful if you have cheats enabled.

**Comment by Gymnasiast** [comment]
Longer stations allow fitting more vehicles in the regular continuous circuit mode, which is useful without the cheat.
Not that that even matters, though, as we don't prevent massive brake runs, unnecessary chain lifts, high speed unbanked curves, and more bad designs either. That's not in the spirit of classic RCT.

**Comment by Broxzier** [comment]
I agree with @Gymnasiast. There's no limit on any other track parts, so why should there be on stations?

**Comment by IntelOrca** [comment]
@Gymnasiast I am just wary as to whether very long stations, like the size of the map could cause the game to go wrong. It might also have performance implications, particularly when enabling synchronisation.

**Comment by jensj12** [comment]
There seems to be an issue with max trains/lenght calculation:
scr101

**Comment by Gymnasiast** [comment]
@jensj12 That must be an outdated version of OpenRCT2, as it still uses dropdowns, rather than spinners.

**Comment by Gymnasiast** [comment]
@IntelOrca That would be at least a proper reason to limit it. I don't know if it causes problems, but let's settle for 32 then. (For now, in any case.)

**Comment by Wirlie** [comment]
> The limit is completely arbitrary and does not make the game more challenging.

I'm not agree with this, this influences the challenging of the game in some way when designing an attraction. Why? Because when designing an attraction you should take into account the limit of the stations length, this forces us to use other methods such as the "block brakes" to ensure that we have the largest number of trains possible.

Instead of a general change, I prefer this with a cheat configuration.
(I tried to explain this in the best way possible, sorry for my mistakes.)

PS: I have the same issue of @jensj12, I cannot set the train length more than four cars:

scr9

**Comment by janisozaur** [comment]
@Wirlie are you sure you have the cheat enabled?

**Comment by Broxzier** [comment]
With miniature railways it's even worse. When the station exceeds 15 tiles in length, it only allows for 1 car per train, which means no guest can ride it. With 15-tile long station I can select 6 cars per train, and with a 16-tile long station I can only select 1 car per train, even with just 1 train selected.

**Comment by Wirlie** [comment]
No, I do not have the cheat enabled. The station length limit should be modified only if the cheat is enabled (as you said: https://github.com/OpenRCT2/OpenRCT2/pull/3486#issuecomment-216649799), because this seems not to work well with the calculation of the max trains and vehicles

**Comment by jensj12** [comment]
Someone on the forums did some more testing: https://openrct2.org/forums/topic/1235-increase-station-platform-length-from-12-to-14/#comment-6256

**Comment by Broxzier** [comment]
I have the feeling there is a `station_length & 0xF` used for those checks..

**Comment by Gymnasiast** [comment]
There is indeed, in `src/ride/ride.c`. I'm going to take that out and test it again.

**Comment by KKLD** [comment]
Hi can we set the limit to 14 for now and merge it?
No limit at all is the best but it doesn't work as of right now and I assume it will take a while to fix.

**Comment by Gymnasiast** [comment]
@KKLD: I have indentified the probable cause. It will probably be fixed over the weekend. If it takes longer I might opt for 14 in the meantime.

**Comment by KKLD** [comment]
@Gymnasiast Okay, cheers :+1:

**Comment by Gymnasiast** [comment]
Ok, I changed my PR to only raise the limit to 32. I also got rid of the faulty length check, as well as a small papercut involving the vehicle window incorrectly displaying singular strings on opening.

File: src/ride/track.c [hunk 1]
```diff
@@ -4186,7 +4186,7 @@ static bool track_add_station_element(int x, int y, int z, int direction, int ri
 		return false;
 	}
 
-	if (stationLength > 12) {
+	if (stationLength > 32) {
```
Comment (IntelOrca) [inline src/ride/track.c]: A global constant would be good as it will be easier to change.

**Comment by Gymnasiast** [comment]
Ok, changed the code to use a define.

**Comment by KKLD** [comment]
May I ask why there is a need for a limit? I mean of course not many people would get close to 32 but someone might want to be able to make a longer station platform and I think they should have the liberty to do so. I am not a fan of being limited :D

And this makes me think about the 31 cars limit for rides such as Go Karts, Car Ride, Log Flume, Splash Boats, all boat-rides and many more. Could we bypass this limit as well?

**Comment by Gymnasiast** [comment]
It might cause problems somewhere else in the code. Personally I wanted no limit either, but we stuck with 32 for now because IntelOrca expected fewer problems. The limit isn't set in stone however, we might raise it again.

As for 31 cars: I think that's an SV6 limitation, but I will look into it.

**Comment by IntelOrca** [comment]
This is a trial run, if there are no problems reported with 32 length stations, we can increase it further.

**Pull request merged**